server/api/src/rsDataObjOpen.cpp

```diff
@@ -350,9 +357,38 @@ namespace
                 //   4. Insert new replica into RST
                 // Should anything go wrong with these steps, we would also require an unlock interface which
                 // does not require a target replica (in case something goes wrong with the registration itself.
-                if (const auto ret = check_if_data_object_is_locked(_inp, ir::make_replica_proxy(*_existing_replica_list)); ret < 0) {
+                if (const auto ret = check_if_data_object_is_locked(_inp, id::make_data_object_proxy(*_existing_replica_list), hierarchy); ret < 0) {
+                    freeL1desc(l1_index);
+
                     return ret;
                 }
+
+                const auto obj = id::make_data_object_proxy(*_existing_replica_list);
+                if (const auto insert_ec = rst::insert(obj); insert_ec < 0) {
+                    freeL1desc(l1_index);
+
+                    return insert_ec;
+                }
+
+                if (const int lock_ec = ill::lock_before_create_and_publish(_comm, obj.data_id(), ill::lock_type::write); lock_ec < 0) {
+                    irods::log(LOG_ERROR, fmt::format(
+                        "[{}:{}] - Failed to lock data object on create "
+                        "[error_code=[{}], path=[{}], hierarchy=[{}]",
+                        __FUNCTION__, __LINE__, lock_ec, _inp.objPath, hierarchy));
+
+                    if (const int unlock_ec = ill::unlock_before_create_and_publish(_comm, obj.data_id(), ill::restore_status); unlock_ec < 0) {
+                        irods::log(LOG_ERROR, fmt::format(
+                            "Failed to unlock data object "
+                            "[error_code={}, path={}, hierarchy={}]",
```

Comment:
this is a different format... need more [] i think

Reply:
Done